governance-aepp/src/utils/aeternity.js

```diff
@@ -89,6 +90,33 @@ aeternity.initStaticClient = async () => {
   });
 };
 
+aeternity.verifyPollContract = async (pollAddress) => {
+  const contractCreateBytecode = await fetch(`${settings[aeternity.networkId].middlewareUrl}/middleware/contracts/transactions/address/${pollAddress}?limit=1&page=1`).then(async res => {
+    const contractCreateTx = (await res.json()).transactions.filter(tx => tx.tx.type === 'ContractCreateTx')[0];
+    return contractCreateTx ? contractCreateTx.tx.code : null;
+  })
+
+  const compilersResult = await Promise.all(settings.compilers.map(compiler => {
+    return fetch(`${compiler.url}/compile`, {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({
+        code: pollContractSource,
+        options: { 'backend': 'fate' }
+      })
+    }).then(async res => {
```

Comment:
maybe we could await the middleware query here since we wait for all compilers anyways

Reply:
yes

Reply:
adjusted in https://github.com/aeternity/aepp-governance/pull/131/commits/18db6ba90e9a49b1115f62d18f8866622b74de85